### PR TITLE
[WIP] Feature Truncate Component & Filter

### PR DESCRIPTION
**Pull request by cdcabrera**
Truncate component and filter

## ToDo
- [x] Initial port from [Origin-Web-Common](https://github.com/cdcabrera/origin-web-common/blob/master/src/components/truncate-long-text/truncateLongText.js)
- [ ] Unit Tests Updated
- [ ] Review need for transcluded content to avoid additional filters
- [ ] Add in appropriate CSS references

### CODE REVIEW

File: src/truncate/truncate.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<span>
+  <span ng-if="!$ctrl.expanded && ctrl.truncated">
+    <span ng-attr-title="{{$ctrl.content}}" class="">
+      <span ng-bind-html="$ctrl.truncatedContent"></span>&hellip;
+    </span>
+    <a ng-href="" ng-click="$ctrl.expanded = true" class="nowrap">{{$ctrl.expandCopy||'See All'}}</a>
```
Comment (jeff-phillips-18) [inline src/truncate/truncate.html]: Is the "nowrap" class defined?

File: src/truncate/truncate.component.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    newlineLimit: '<',
+    removeExtraChars: '<',
+    expandCopy: '<',
+    collapseCopy: '<',
+    // When expandable is on, optionally hide the collapse link so text can only be expanded. (Used for toast notifications.)
+    expandable: '<',
+    //hideCollapse: '<',
+    // optional keywords to highlight using the `highlightKeywords` filter
+    keywords: '<',
+    prettifyJson: '<'
```
Comment (cdcabrera) [inline src/truncate/truncate.component.js]: It was part of the original port from [Origin-Web-Common](https://github.com/cdcabrera/origin-web-common/blob/master/src/components/truncate-long-text/truncateLongText.js#L18), currently it doesn't do anything. Was debating on whether to even include it or come up with some form of transclusion for that type of content...

File: src/truncate/truncate.component.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    useWordBoundary: '<',
+    newlineLimit: '<',
+    removeExtraChars: '<',
+    expandCopy: '<',
+    collapseCopy: '<',
+    // When expandable is on, optionally hide the collapse link so text can only be expanded. (Used for toast notifications.)
+    expandable: '<',
+    //hideCollapse: '<',
+    // optional keywords to highlight using the `highlightKeywords` filter
+    keywords: '<',
```
Comment (cdcabrera) [inline src/truncate/truncate.component.js]: Currently it isn't used within this component... part of the original port from [Origin-Web-Common](https://github.com/cdcabrera/origin-web-common/blob/master/src/components/truncate-long-text/truncateLongText.js). Debating whether to include it, or simply allow some form of transcluded content

File: src/truncate/truncate.html (hunk 1 above)
Comment (cdcabrera) [inline src/truncate/truncate.html]: Class currently is not defined, part of the original port. Clean up still required

**Comment by cdcabrera** [comment]
closing this down until the related [origin-web-common PR 107 gets finalized](https://github.com/openshift/origin-web-common/pull/107)

**Pull request closed without merging**